### PR TITLE
Fix cmake

### PR DESCRIPTION
**Pull request by mxswd**
Fixed LIBEDIT failure, not enough global mutable state for CMake!

### CODE REVIEW

**Comment by mxswd** [comment]
@jrose-apple what do you think?

**Comment by shahmishal** [comment]
@swift-ci Please test

File: CMakeLists.txt [hunk 1]
```diff
@@ -13,6 +13,7 @@ set_property(GLOBAL PROPERTY JOB_POOL_LINK local_jobs)
 
 # First include general CMake utilities.
 include(SwiftUtils)
+include(CheckSymbolExists)
```
Comment (jrose-apple) [inline CMakeLists.txt]: Nitpick: this belongs down with the other `include(Check*)` lines (where you put the new one).

**Review by jrose-apple — APPROVED** [review]
If it works it looks good to me. @erg may want to check it out too.

([Docs for `cmake_push_check_state`](https://cmake.org/cmake/help/v3.0/module/CMakePushCheckState.html))

**Comment by jrose-apple** [comment]
Also cc @hughbe, just to keep him in the loop.

**Comment by CodaFi** [comment]
@swift-ci please clean test.

**Comment by jrose-apple** [comment]
```
/home/buildnode/jenkins/workspace/swift-PR-Linux/branch-master/swift/lib/Immediate/REPL.cpp:365:7: error: use of undeclared identifier 'wcslcat'; did you mean 'wcsncat'?
```

Looks like our Ubuntu actually has a libedit. Maybe we *can* port the REPL there?

**Comment by mxswd** [comment]
Could we just change it to `wcsncat`?

**Comment by jrose-apple** [comment]
Yes, I'm happy to take that change (along with any other necessary fix-up if we're relying on the l-ness of wcslcat).

**Comment by mxswd** [comment]
`wcsncat` takes an l? (http://www.cplusplus.com/reference/cwchar/wcsncat/)

**Comment by jrose-apple** [comment]
BSD extensions FTW!

```
     The strlcpy() and strlcat() functions copy and concatenate strings
     respectively.  They are designed to be safer, more consistent, and less
     error prone replacements for strncpy(3) and strncat(3).  Unlike those
     functions, strlcpy() and strlcat() take the full size of the buffer (not
     just the length) and guarantee to NUL-terminate the result (as long as
     size is larger than 0 or, in the case of strlcat(), as long as there is
     at least one byte free in dst).  Note that a byte for the NUL should be
     included in size.  Also note that strlcpy() and strlcat() only operate on
     true ``C'' strings.  This means that for strlcpy() src must be NUL-termi-
     nated and for strlcat() both src and dst must be NUL-terminated.

     The strlcpy() function copies up to size - 1 characters from the NUL-ter-
     minated string src to dst, NUL-terminating the result.

     The strlcat() function appends the NUL-terminated string src to the end
     of dst.  It will append at most size - strlen(dst) - 1 bytes, NUL-termi-
     nating the result.
```

**Comment by jrose-apple** [comment]
@swift-ci Please clean test

**Comment by jrose-apple** [comment]
@swift-ci Please clean test

**Comment by jrose-apple** [comment]
And again to make it clean, avoiding the Wrong SHA failure.

@swift-ci Please clean test

File: CMakeLists.txt [hunk 2]
```diff
@@ -805,6 +807,17 @@ if(LIBXML2_FOUND)
...
 
+# You need libedit linked in order to check if you have el_wgets.
+cmake_push_check_state()
+list(APPEND CMAKE_REQUIRED_LIBRARIES "edit")
+check_symbol_exists(el_wgets "histedit.h" HAVE_EL_WGETS)
+check_symbol_exists(wcslcat wchar.h HAVE_WCSLCAT)
+# FIXME: UNICODE_LIBEDIT shouldn't need BSD's wcslcat function,
+#        rewrite the usages in terms of wcsncat to get the REPL
+#        building on linux.
+set(HAVE_UNICODE_LIBEDIT (HAVE_EL_WGETS AND HAVE_WCSLCAT))
```
Comment (jrose-apple) [inline CMakeLists.txt]: Oh. Hang on. CMake doesn't do this. You need to use `if`. ><

Comment (mxswd) [inline CMakeLists.txt]: Oh...

**Comment by jrose-apple** [comment]
@swift-ci Please clean test

**Pull request merged**